crates/astria-sequencer-relayer/src/transaction.rs

```diff
@@ -1,24 +1,37 @@
-use prost;
 use tendermint::{
     hash::Hash as TmHash,
     merkle,
 };
 
 use crate::base64_string::Base64String;
 
-fn tx_to_prost_bytes(tx: Vec<u8>) -> prost::alloc::vec::Vec<u8> {
-    let mut buf = prost::alloc::vec::Vec::new();
-    prost::encoding::bytes::encode(1, &tx, &mut buf);
-    buf
-}
-
 pub fn txs_to_data_hash(txs: &[Base64String]) -> TmHash {
-    let txs = txs
-        .iter()
-        .map(|tx| tx_to_prost_bytes(tx.0.clone()))
-        .collect::<Vec<Vec<u8>>>();
+    let txs = txs.iter().map(|tx| tx.0.clone()).collect::<Vec<Vec<u8>>>();
 
     TmHash::Sha256(merkle::simple_hash_from_byte_vectors::<
         tendermint::crypto::default::Sha256,
     >(&txs))
 }
+
+#[cfg(test)]
+mod test {
+    use sha2::Digest;
+
+    use super::*;
+
+    #[test]
+    fn txs_to_data_hash_test() {
+        // data_hash is calculated from the txs in a block, where the leaves of the merkle tree are
+        // the sha256 hashes of the txs
+        let tx = Base64String::from_string("CscBCsQBCg0vU2VxdWVuY2VyTXNnErIBCghldGhlcmV1bRJ4Avh1ggU5gIRZaC8AhQUD1cTyglIIlBtwp0/22gQLMRmQwVX9/9u8AvfuiA3gtrOnZAAAgMABoLnRqksJblEaolE6wbsAHYTAiSlA14+B5nvWuFrIfevnoBg+UGcWLC4eg1lZylqLnrL8okBc3vTS4qOO/J5sRtVDGixtZXRybzFsbDJobHAzM3J4eTdwN2s2YXhoeDRjdnFtdGcwY3hkZjZnemY5ahJ0Ck4KRgofL2Nvc21vcy5jcnlwdG8uc2VjcDI1NmsxLlB1YktleRIjCiEDJ/LvaMZTBcGX66geJOEmTm/fyyPTZKMUJoDtMDUmSPkSBAoCCAESGAoQCgV1dGljaxIHMTAwMDAwMBCAlOvcAyIIZXRoZXJldW0aQMhoTCUr84xgTkYxsFWDfHH2k+oHCPsKvbTpz8m5YrHfYMv6gdou6V8oj1v0B9ySD5VjMXQi1kJ9DZN6wD2buo8=".to_string()).unwrap();
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(tx.0);
+        let hash = hasher.finalize();
+
+        let expected_hash =
+            Base64String::from_string("rRDu3aQf1V37yGSTdf2fv9GSPeZ6/p0wJ9pjBl8IqFc=".to_string())
```

Comment:
Not a fan of these magic number tests. What does it tell us should they ever fail? It's very inscrutable.

I don't know if this can be easily solved in a unit test. Maybe spin up an instance of the thing that's supposed to understand this message?

Reply:
yeah, the `header_verify_hashes` should be testing this, except the problem is we also need to submit a transaction to validate a data_hash that isn't just the empty merkle tree hash, but there's no tx signing library in rust for metro :| when we move to the rust sequencer, this will definitely be doable!